frontend/src/components/DataProductUploader.vue

```diff
@@ -0,0 +1,318 @@
+        <span class="upload-input__text">{{ file ? file.name : 'Drag and drop your file or click to browse' }}</span>
+        <span class="upload-size__text">Max. file size 2 GB</span>
+        <input class="file-upload__input" type="file" @change="onFileSelected" ref="fileInput"/>
+      </label>
+      <div style="display: flex; justify-content: center;">
+        <div class="text-label">
+          <span>Or</span>
+        </div>
+      </div>
+      <label class="input__label">
+        <span class="upload-input__text">
+          <img class="upload-input__icon" src="../assets/icons/fill-up-form.svg"/>
+          Fill up a form
+        </span>
+        <input class="file-upload__input" type="button" @click="fillUpForm"/>
+      </label>
+      <div class="upload-container">
+        <div style="display: flex; justify-content: center;">
+          <div class="text-label">
+            <span>To</span>
+          </div>
+        </div>
+        <div class="upload-checkboxes" :class="{ 'no-click': uploading}">
+          <label>
+            <input type="radio" v-model="selectedDatabase" value="Literature Database">
+            <span class="checkbox-label" :class="{ 'checkbox-selected': selectedDatabase === 'Literature Database' }">Literature Database</span>
+          </label>
+          <label>
+            <input type="radio" v-model="selectedDatabase" value="Computational Database">
+            <span class="checkbox-label"
+                  :class="{ 'checkbox-selected': selectedDatabase === 'Computational Database' }">Computational Database</span>
+          </label>
+          <label>
+            <input type="radio" v-model="selectedDatabase" value="Experimental Database">
+            <span class="checkbox-label" :class="{ 'checkbox-selected': selectedDatabase === 'Experimental Database' }">Experimental Database</span>
+          </label>
+        </div>
+        <div class="progress-bar-container" v-if="progress !== 0">
+          <div class="file-upload__progress" v-if="uploading">
+            <div class="file-upload__progress-bar" :style="{ width: progress + '%' }"/>
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="file-upload__footer">
+      <div class="mr-3">
+        <button class="discard_button" :class="{'discard_button--disabled' : !file || uploading}"
+                :disabled="!file || uploading" @click="discard">Discard
+        </button>
+        <button class="btn btn-primary" :disabled="!file || uploading" @click="uploadFile">Next</button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import axios from 'axios';
+
+export default {
+  name: "FileUpload",
+
+  data() {
+    return {
+      file: null,
+      uploading: false,
+      selectedDatabase: 'Computational Database',
+      formOpen: false,
+      progress: 0
+    };
+  },
+
+  computed: {
+    uploadUrl() {
+      const URL_PREFIX = 'http://127.0.0.1:8000/smiles/';
```

Comment:
Change this to `'/smiles/'`. With the webpack loader integration, you won't need this any longer. Also, you shouldn't need any special CORS configuration.